### PR TITLE
feat: Docker-compose file to run the project locally

### PR DESCRIPTION
**Pull request by khorolets**
This is an initial version of the `docker-compose.yml` for running the project locally.

I've left a "TODO" section about the Docker compose in the README with the things I want us to improve in the future:

- Setup all the services to use volumes for the codebase and use `cargo watch` to recompile the code on the fly
- Come up with a way of changing the configuration for the indexers in a convenient way
- Robust documentation for the docker-compose setup

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+version: '3'
+
```
Comment (kobayurii) [inline docker-compose.yml:2]: I see you are duplicating changes. I propose to put all the common variables in the 

x-common-variables: &common-variables
  SCYLLA_URL: scylla:9042
      SCYLLA_USER: ${SCYLLA_USER}
      SCYLLA_PASSWORD: ${SCYLLA_PASSWORD}
      AWS_BUCKET_NAME: near-lake-data-mainnet
      NEAR_RPC_URL: https://archival-rpc.mainnet.near.org
      AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID}
      AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}
      AWS_DEFAULT_REGION: ${AWS_DEFAULT_REGION}
      OTEL_EXPORTER_JAEGER_AGENT_HOST: jaeger
      OTEL_EXPORTER_JAEGER_AGENT_PORT: 6831

I'm not sure if version 3 of compose supports this use case. Maybe you need to change it to 3.4 or higher

Translated with www.DeepL.com/Translator (free version)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+services:
+  rpc-server:
+    build:
+      context: .
+      dockerfile: rpc-server/Dockerfile
+      args:
+        features: "tracing-instrumentation,shadow_data_consistency"
+    environment:
+      SCYLLA_URL: scylla:9042
```
Comment (kobayurii) [inline docker-compose.yml:11]: use common-variables 
    environment:
      <<: *common-variables
     SERVER_PORT: 8080

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      - jaeger
+
+  state-indexer:
+    build:
+      context: .
+      dockerfile: state-indexer/Dockerfile
+      args:
+        features: tracing-instrumentation
+    environment:
+      INDEXER_ID: state-indexer-local
```
Comment (kobayurii) [inline docker-compose.yml:37]: use common-variables 
    environment:
      <<: *common-variables

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    restart: on-failure
+
+  tx-indexer:
+    build:
+      context: .
+      dockerfile: tx-indexer/Dockerfile
+      args:
+        features: tracing-instrumentation
+    environment:
+      INDEXER_ID: tx-indexer-local
```
Comment (kobayurii) [inline docker-compose.yml:59]: use common-variables 
    environment:
      <<: *common-variables

**Review by kobayurii — APPROVED** [review]

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    ports:
+      - 5775:5775/udp
+      - 6831:6831/udp
+      - 6832:6832/udp
+      - 5778:5778
+      - 16686:16686
+      - 14268:14268
+      - 9411:9411
+
+  scylla:
```
Comment (kobayurii) [inline docker-compose.yml:85]: Do we need a `volumes` for the database?

**Review by kobayurii — APPROVED** [review]

File: docker-compose.yml (hunk 5 above)
Comment (khorolets) [reply docker-compose.yml:77]: Not by default. If developers need one, they could add it manually.

**Pull request merged**